packages/lockfile-lint-api/src/validators/ValidateIntegrity.js

```diff
@@ -0,0 +1,54 @@
+'use strict'
+
+module.exports = class ValidateIntegrity {
+  constructor ({packages} = {}) {
+    if (typeof packages !== 'object') {
+      throw new Error('expecting an object passed to validator constructor')
+    }
+
+    this.packages = packages
+  }
+
+  validate () {
+    let validationResult = {
+      type: 'success',
+      errors: []
+    }
+
+    for (const [packageName, packageMetadata] of Object.entries(this.packages)) {
+      if (!('integrity' in packageMetadata)) {
+        continue
+      }
+
+      try {
```

Comment:
about the integrity check here - 
the registry uses SHA512 for tarballs, but if a source of a package is provided to be git, then it defaults to the commit hash, which is a SHA1. We'd need a case here to allow those type of sources to not fail the validation check as you'd otherwise have a conflict that can't be resolved by the user.

Reply:
Can you give an example?

From what I saw, if you install a dependency directly from a GitHub link, the `package-lock.json` file does not have the `integrity` field for that dependency (only for the transitive dependencies). It either has a version field with a link with hash, or the resolved field with link with hash. My validator currently skips dependencies that don't have the `integrity` field

Example from "lockfileVersion": 2
```
    "eslint": {
      "version": "git+ssh://git@github.com/eslint/eslint.git#bf7bd885a92046a6b6bcbcaaa1e78e9f2c4b482f",
      "from": "eslint@github:eslint/eslint",
...
    "node_modules/eslint": {
      "version": "8.23.1",
      "resolved": "git+ssh://git@github.com/eslint/eslint.git#bf7bd885a92046a6b6bcbcaaa1e78e9f2c4b482f",

```

Reply:
What you described is what I meant in `then it defaults to the commit hash, which is a SHA1` :-)
I think we're good here if the code skips no integrity at all.